### PR TITLE
Add Solana Transaction Conversation

### PR DESCRIPTION
**Pull request by enoldev**

### CODE REVIEW

File: sol-transactions/convo.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+			return loop.Quit(fmt.Errorf("invalid start block input value %q, expected a number", msg.Value))
+		}
+
+		c.state.InitialBlock = initialBlock
+		c.state.InitialBlockSet = true
+		return c.NextStep()
+
+	case AskProgramId:
+		return c.action(InputProgramId{}).
+			TextInput(fmt.Sprintf("Filter the transactions based on one or several Program IDs.\n\nSupported operators are: logical or '||', logical and '&&' and parenthesis: '()'. \n\nExample: to only consume TRANSACTIONS containing Token or ComputeBudget instructions: 'program:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA || program:ComputeBudget111111111111111111111111111111'. \n\nTransactions containing 'Vote111111111111111111111111111111111111111' instructions are always excluded."), "Submit").
```
Comment (abourget) [inline sol-transactions/convo.go]: This language is common across all of our modules.

@enoldev could we have a well formatted documentation page that explains all the formats and syntax of that language.. and then link from here?

File: sol-transactions/convo.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+			downloadCmd.AddFile(fileName, fileContent, "text/plain", fileDescription)
+		}
+
+		return loop.Seq(c.msg().Messagef("Code generation complete!").Cmd(), downloadCmd.Cmd())
+
+	case codegen.InputSourceDownloaded:
+		return c.NextStep()
+
+	case codegen.RunBuild:
```
Comment (abourget) [inline sol-transactions/convo.go]: we're not doing remote builds anymore.. so perhaps we can simplify here?

File: sol-transactions/generate.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+	projFiles, err = p.Render()
+	if err != nil {
+		return nil, fmt.Errorf("rendering template: %w", err)
+	}
+
+	return
+}
+
+func (p *Project) build(remoteBuildContentChan chan<- *codegen.RemoteBuildState) {
```
Comment (abourget) [inline sol-transactions/generate.go:86]: lots of duplications for remote building here.. let's purge, in any case, when we re-enable, it's most certainly going to be abstracted away, not duplicated 7 times like right now

File: sol-transactions/templates/README.md.gotmpl [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Solana Transactions
+
+This Substreams project allows you to retrieve Solana transactions filtered by one or several Program IDs (i.e. you will only receive transactions containing the specified Program IDs).
+**NOTE:** Transactions containing voting instructions will NOT be present.
+
+## Understand the Generated Project
+
+Only a `substreams.yaml` file has been generated. This file declare a Substreams module, `map_filtered_transactions`, which uses the Solana Foundational Modules (module built by the team).
```
Comment (enoldev) [inline sol-transactions/templates/README.md.gotmpl]:
```suggestion
Only a `substreams.yaml` file has been generated. This file declares a Substreams module, `map_filtered_transactions`, which uses a Solana Foundational Module (a module built by the team).
```

File: sol-transactions/templates/README.md.gotmpl [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+params:
+  map_filtered_transactions: {{ .ProgramId }} // 4.
+```
+1. Import the Solana Foundational Modules.
+2. Declare the `map_filtered_transactions` module, which you will run later.
+3. Use the `filtered_transactions_without_votes` module from the Solana Foundational Modules.
+Essentially, you are _using_ the Solana Foundational Module, which is pre-built for you.
+4. Pass the regular expression to filter the transactions based on the specified Program IDs.
+
+## Authenticate
```
Comment (abourget) [inline sol-transactions/templates/README.md.gotmpl]: These "next steps" commands should be first up there, with only the commands like:
```bash
substreams build
substreams auth
substreams gui
```

as a continuation of your onboarding.

And if this module leads an easy way to subgraphs, thne we can add: `substreams codegen subgraph`

File: sol-transactions/templates/README.md.gotmpl [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+To run your Substreams you will need to [authenticate](https://substreams.streamingfast.io/documentation/consume/authentication) yourself.
+
+```bash
+substreams auth
+```
+
+## Run your Substreams
+
+```bash
```
Comment (abourget) [inline sol-transactions/templates/README.md.gotmpl]: The `substreams gui` now needs _zero_ parameters.. the endpoint is selected automatically.. start block is set automatically from the top-most module (in the ancestor tree), stop block is +1000 by default.

File: sol-transactions/convo.go (hunk 1 above)
Comment (enoldev) [reply sol-transactions/convo.go]: The syntax is specific to every foundational module, do you mean a document with instructions for every foundational module?

Comment (sduchesneau) [reply sol-transactions/convo.go]: @enoldev , no, the syntax is generic and used in any module using a filter.

the parenthesis, the && and the || ...

Comment (sduchesneau) [reply sol-transactions/convo.go]: But since this is on a Prompt, it makes sense to print it all out here, so I'm good with this code.

**Pull request merged**